packages/@o3r/rules-engine/src/engine/ruleset-executor.ts

```diff
@@ -195,38 +195,62 @@ export class RulesetExecutor {
     throw new Error(`Unknown condition block met : ${JSON.stringify(nestedCondition)}`);
   }
 
+  // TODO: make public and move to top of the class in a separate PR to make the diff more meaningful in this one
   /**
    * Plug ruleset to fact streams and trigger a first evaluation
-   *
-   * @param ruleset
    */
   public prepareRuleset() {
-    const rulesWithContext: Rule[] = [];
-    const rulesWithoutContext: Rule[] = [];
+    const inputFactsForRule: Record<string, string[]> = {};
+    this.ruleset.rules.forEach((rule) => {
+      const findRuleInputFacts = (obj: AllBlock): string[] => {
+        const ruleInputFacts = new Set<string>();
+        // Recursive function to explore each object
+        const exploreObject = (currentObj: any) => {
```

Comment:
[not mandatory] 
Maybe this method could be defined outside to avoid redefining it for every rules

Reply:
This is actually mandatory for my point of view, for performance purpose